script.js

```diff
@@ -157,7 +157,12 @@ function touchingEdge(direction, snakeBlocks) {
   } else if (direction == "down") {
     newPos = [snakeBlocks[snakeBlocks.length - 1][0], snakeBlocks[snakeBlocks.length - 1][1] + 1];
   }
-  if (newPos[0] >= 30 || newPos[0] < 0 || newPos[1] >= 30 || newPos[1] < 0) {
+  return newPos
```

Comment:
Given that all the branches of your `if`/`else` chain all set `newPos` to then be returned here, I'd write it to just return directly, e.g.:

```js
if (direction == "right") {
    return [snakeBlocks[snakeBlocks.length - 1][0] + 1, snakeBlocks[snakeBlocks.length - 1][1]];
}
```

It just makes it easier to see what's going on, the whole function then consists of four branches all of which return immediately so I don't have to worry about what's going to happen to `newPos` after it's set.